mundane-parse/src/main/scala/com/ambiata/mundane/parse/ListParser.scala

```diff
@@ -273,13 +273,12 @@ object ListParser {
    */
   def pair[A, B](pa: ListParser[A], pb: ListParser[B], delimiter: Char): ListParser[(A, B)] =
     ListParser((position, state) => state match {
-      case h :: t   =>
+      case h :: t =>
         Delimited.parseRow(h, delimiter) match {
           case a :: b :: Nil =>
-            pa.parse(List(a)).flatMap { case (_, _, ra) => pb.parse(List(b)).map { case (_, _, rb) => (position + 1, t, (ra, rb)) } }
+            (pa.run(List(a)) |@| pb.run(List(b)))((a,b) => (position + 1, t, a -> b )).leftMap(e => position -> e)
```

Comment:
Not that it matters, but I'm not sure this necessarily helps given that `Monad[ListParser]` only implements point/bind.